src/avx512-common-qsort.h

```diff
@@ -671,14 +671,97 @@ static inline int64_t partition_avx512(type_t1 *keys,
 }
 
 template <typename vtype, typename type_t>
-void qsort_(type_t *arr, int64_t left, int64_t right, int64_t maxiters);
+X86_SIMD_SORT_INLINE type_t get_pivot(type_t *arr,
+                                      const int64_t left,
+                                      const int64_t right);
+
+template <typename vtype, int64_t maxN>
+X86_SIMD_SORT_INLINE void sort_n(typename vtype::type_t *arr, int N);
 
 template <typename vtype, typename type_t>
-void qselect_(type_t *arr,
-              int64_t pos,
-              int64_t left,
-              int64_t right,
-              int64_t maxiters);
+static void qsort_(type_t *arr, int64_t left, int64_t right, int64_t max_iters)
+{
+    /*
+     * Resort to std::sort if quicksort isnt making any progress
+     */
+    if (max_iters <= 0) {
+        std::sort(arr + left, arr + right + 1);
+        return;
+    }
+    /*
+     * Base case: use bitonic networks to sort arrays <= vtype::network_sort_threshold
+     */
+    if (right + 1 - left <= vtype::network_sort_threshold) {
+        sort_n<vtype, vtype::network_sort_threshold>(
+                arr + left, (int32_t)(right + 1 - left));
+        return;
+    }
+
+    type_t pivot = get_pivot<vtype, type_t>(arr, left, right);
+    type_t smallest = vtype::type_max();
+    type_t biggest = vtype::type_min();
+
+    int64_t pivot_index;
+
+    if constexpr (vtype::partition_unroll_factor != 0) {
```

Comment:
Is there a reason why we need this if condition? `partition_avx512_unrolled` calls `partition_avx512` if under this condition.